.github/workflows/test-lang-py-ARM.yml

```diff
@@ -43,8 +43,18 @@ jobs:
 
       - name: Install dependencies
         run: |
+          set -xe
           sudo apt-get update -q
-          sudo apt-get install -q -y python3 python3-dev python3-pip git libbz2-dev libjansson-dev liblzma-dev libsnappy-dev libzstd-dev
+          sudo apt-get install -y software-properties-common
+          sudo add-apt-repository -y ppa:deadsnakes/ppa
+          sudo apt-get update -q
+          sudo apt-get install -q -y python3.12 python3.12-dev python3-pip git libbz2-dev libjansson-dev liblzma-dev libsnappy-dev libzstd-dev
+          sudo update-alternatives --install /usr/bin/python python /usr/bin/python3.8 1
+          sudo update-alternatives --install /usr/bin/python python /usr/bin/python3.12 2
+          sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.8 1
+          sudo update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.12 2
+          sudo update-alternatives  --set python /usr/bin/python3.12
+          sudo update-alternatives  --set python3 /usr/bin/python3.12
           python3 -m pip install --upgrade pip setuptools tox
```

Comment:
Wuff, I don't know what version of pip `python3-pip` gets us, but [the code that caused the failure](https://github.com/apache/avro/actions/runs/12547916868/job/34986217286?pr=3221#step:4:220) was removed in https://github.com/pypa/pip/commit/6739f56351a88add7a9e09e4eec25f691f79ec79, which was released in v21.0 (the first release of 2021).

Anyway, I guess to work around it we need to update pip using a version of python that is before 3.12. IIUC, python3.8 is still here, so `python3.8 -m pip install --upgrade pip setuptools tox`.

(For now, at least, [the latest pip](https://pip.pypa.io/en/stable/news/#v24-3-1) is still compatible with python 3.8, or we might need to get more boot-strappy.)